karoo_gp/engine.py

```diff
@@ -134,6 +185,14 @@ def __init__(self, model, tf_device="/gpu:0", tf_device_log=False):
             allow_soft_placement=True)
         self.config.gpu_options.allow_growth = True
 
+        def safe_sqrt(a):
+            """For sqrt(-a), return -sqrt(abs(a))"""
+            negative = tf.less(a, 0)
+            absolute = tf.abs(a)
+            square_root = tf.sqrt(absolute)
+            square_root[negative] *= -1
+            return square_root
```

Comment:
This one uses `tensorflow` instead of `numpy`, and it's defined within `__init__` so it doesn't take self. It was done this way to avoid having to load `tensorflow` unless/until you ask for it.